plugins/prometheus_plugin/prometheus_plugin.cpp

```diff
@@ -1,19 +1,168 @@
 #include <eosio/prometheus_plugin/prometheus_plugin.hpp>
 #include <fc/log/logger.hpp>
+#include <prometheus/metric_family.h>
+#include <prometheus/collectable.h>
+#include <prometheus/counter.h>
+#include <prometheus/summary.h>
+#include <prometheus/text_serializer.h>
+#include <prometheus/registry.h>
+
+namespace prometheus {
+   std::vector<MetricFamily> CollectMetrics(
+         const std::vector<std::weak_ptr<prometheus::Collectable>>& collectables) {
+      auto collected_metrics = std::vector<MetricFamily>{};
+
+      for (auto&& wcollectable : collectables) {
+         auto collectable = wcollectable.lock();
+         if (!collectable) {
+            continue;
+         }
+
+         auto&& metrics = collectable->Collect();
+         collected_metrics.insert(collected_metrics.end(),
+                                  std::make_move_iterator(metrics.begin()),
+                                  std::make_move_iterator(metrics.end()));
+      }
+
+      return collected_metrics;
+   }
+
+   class MetricsHandler {
+   public:
+      std::mutex collectables_mutex_;
+      std::vector<std::weak_ptr<Collectable>> collectables_;
```

Comment:
I don't see any reason to use `weak_ptr` or `shared_ptr` as these will be registered at startup and never removed. We do not unload plugins and I don't see us ever unloading a collection.